apps/zbugs/server/notify.ts

```diff
@@ -71,35 +64,44 @@ export async function notify(
   const modifierUser = await tx.query.user.where('id', modifierUserID).one();
   assert(modifierUser);
 
-  const recipients = await gatherRecipients(
-    // TODO: we shouldn't have to assert and cast this given this is called from `server-mutators`.
-    tx as ServerTransaction<Schema, postgres.TransactionSql>,
-    issueID,
-    isAssigneeChange,
-    previousAssigneeID,
-  );
+  const recipients = await gatherRecipients(tx, issueID, modifierUserID);
 
   // If no recipients, skip notification
   if (recipients.length === 0) {
     console.log('No recipients for notification', issueID);
     return;
   }
 
+  if (issue.shortID === null) {
+    console.log('No short ID for issue', issueID);
+    return;
+  }
+
   // Only send to Discord for public issues
   const shouldSendToDiscord = issue.visibility === 'public';
 
+  const issueLink = `https://bugs.rocicorp.dev/issue/${issue.shortID}`;
+  const getUnsubscribeLink = (email: string) =>
+    `https://bugs.rocicorp.dev/api/unsubscribe?id=${issue.shortID}&email=${encodeURIComponent(email)}`;
+
   switch (kind) {
     case 'create-issue': {
       const payload = {
         title: `${modifierUser.login} reported an issue`,
         message: [issue.title, clip((await issue.description) ?? '')]
           .filter(Boolean)
           .join('\n'),
-        link: `https://bugs.rocicorp.dev/issue/${issue.shortID}`,
+        link: issueLink,
       };
       for (const recipient of recipients) {
         postCommitTasks.push(async () => {
-          await sendEmail({tx, email: recipient, ...payload, issue});
+          await sendEmail({
```

Comment:
There is a lot of duplicated code in here. I know you didn't do it but can you clean it up?